index.bs

```diff
@@ -1477,102 +1501,122 @@ RemoteValue = {
   WindowProxyRemoteValue //
 }
 
+InternalId = String;
```

Comment:
We could make this `ObjectId` in the hope of later reusing it for weakrefs if we add that feature. If we made that optional it would be a bit odd that objects which are part of a cycle always get an objectId. But it would also be odd to have an `internalId` that isn't the same as the `objectId` in case we do take that approach later on.

Reply:
I find using `ObjectId` as `internalId` confusing. From the implementation perspective, if we implement weak `objectId` map, it will be different from the `internalId`, as it will unlikely live in V8.